source-academy/frontend
# Remove deleted files from open editor tabs

### Description

Resolves the following issue that was raised by @zhaojj2209 in #2378:

> - If a file tab is open, the tab will remain even if the file is deleted. Note below that the `/test folder/test.js` tab still exists even though the file no longer exists.
> ![telegram-cloud-photo-size-5-6104838078012175332-y](https://user-images.githubusercontent.com/54243224/226177047-67436b19-99e7-4c85-9857-52884dd60129.jpg)

Part of #2176.

### Type of change


### How to test

The multiple files mode is not yet ready for production and is disabled programmatically. Enable multiple files mode for testing by changing the condition to `false`:
https://github.com/source-academy/frontend/blob/9bc11408b104e29679bfdc025992b241975ea6d7/src/pages/playground/Playground.tsx#L603-L615

Then, verify that deleting files or folders from the file system view results in the corresponding editor tabs being closed.
